File: feeds/receive_notification/java/payment.java
package util;
import java.io.*;
import java.util.*;
import javax.servlet.*;
import javax.servlet.http.*;
import com.mercadopago.MP;

import org.codehaus.jettison.json.JSONObject;
import org.codehaus.jettison.json.JSONArray;

public class ReceiveFeed extends HttpServlet{
 
	public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
		MP mp = new MP ("CLIENT_ID", "CLIENT_SECRET");
		try{
			Map<String, Object> params = new HashMap<String, Object>();
			params.put("access_token", mp.getAccessToken());
			JSONObject merchantOrderInfo=null;
			 
			if(request.getParameter("topic").equals("merchant_order")){
				merchantOrderInfo = mp.get("/merchant_orders/" + request.getParameter("id"), params, false);
			}else if(request.getParameter("topic").equals("payment")){
				JSONObject paymentInfo = mp.get("/collections/notifications/" + request.getParameter("id"), params, false);
				merchantOrderInfo = mp.get("/merchant_orders/" + paymentInfo.getJSONObject("response").getJSONObject("collection").getString("merchant_order_id"), params, false);
			}
			if (merchantOrderInfo.getInt("status") == 200) {
				Double transactionAmountPayments = 0D;
				Double transactionAmountOrder = merchantOrderInfo.getJSONObject("response").getDouble("total_amount");
				JSONArray payments=merchantOrderInfo.getJSONObject("response").getJSONArray("payments");
			    for (int i=0; i < payments.length(); i++) {
			    	JSONObject payment = payments.getJSONObject(i);
			    	if(payment.getString("status").equals("approved")){
			    		transactionAmountPayments += payment.getDouble("transaction_amount");
				    }	
			    }
			    if(transactionAmountPayments >= transactionAmountOrder){
			    	System.out.print("release your items");
			    }else{
					System.out.print("dont release your items");
				}
			}
		}catch(Exception e){
			throw new ServletException(e);
		}
	}
}

File: feeds/receive_notification/php/payment.php
<?php
require_once "../../lib/mercadopago.php";

$mp = new MP("CLIENT_ID", "CLIENT_SECRET");

$params = ["access_token" => $mp->get_access_token()];
if($_GET["topic"] == 'payment'){
	$payment_info = $mp->get("/collections/notifications/" . $_GET["id"], $params, false);
	$merchant_order_info = $mp->get("/merchant_orders/" . $payment_info["response"]["collection"]["merchant_order_id"], $params, false);
}else if($_GET["topic"] == 'merchant_order'){
	$merchant_order_info = $mp->get("/merchant_orders/" . $_GET["id"], $params, false);
}
if ($merchant_order_info["status"] == 200) {
	$transaction_amount_payments= 0;
	$transaction_amount_order = $merchant_order_info["response"]["total_amount"];
    $payments=$merchant_order_info["response"]["payments"];
    foreach ($payments as  $payment) {
    	if($payment['status'] == 'approved'){
	    	$transaction_amount_payments += $payment['transaction_amount'];
	    }	
    }
    if($transaction_amount_payments >= $transaction_amount_order){
    	echo "release your items";
    }
    else{
		echo "dont release your items";
	}
}
?>